pandas/tests/io/test_user_agent.py

```diff
@@ -181,6 +184,47 @@ def do_GET(self):
         self.wfile.write(response_bytes)
 
 
+def wait_until_ready(func, *args, **kwargs):
+    def inner(*args, **kwargs):
+        while True:
+            try:
+                return func(*args, **kwargs)
+            except urllib.error.URLError:
+                time.sleep(0.1)
+
+    return inner
+
+
+def process_server(responder, port):
+    with http.server.HTTPServer(("localhost", port), responder) as server:
+        server.handle_request()
+    server.server_close()
+
+
+@pytest.fixture
+def responder(request):
```

Comment:
can you comment on why all this is necessary

Reply:
Added some comments